packages/design-system/src/components/nested-icon-label.tsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * Implementation of the "Nested Icon Label" component from:
+ * https://www.figma.com/file/sfCE7iLS0k25qCxiifQNLE/%F0%9F%93%9A-Webstudio-Library?node-id=148-3161
+ */
+
+import { forwardRef, type ComponentProps, type Ref } from "react";
+import { type CSS, css, theme } from "../stitches.config";
+import type { labelColors } from "./label";
+
+const colors = {
+  default: {
+    border: "transparent",
+    background: "transparent",
+    backgroundHover: theme.colors.backgroundHover,
+    icon: theme.colors.foregroundIconMain,
+  },
+  preset: {
+    border: theme.colors.borderMain,
+    background: theme.colors.backgroundPresetMain,
+    backgroundHover: theme.colors.backgroundPresetHover,
+    icon: theme.colors.foregroundIconMain,
+  },
+  local: {
+    border: theme.colors.borderSetMain,
+    background: theme.colors.backgroundSetMain,
+    backgroundHover: theme.colors.backgroundSetHover,
+    icon: theme.colors.foregroundSetMain,
+  },
+  remote: {
+    border: theme.colors.borderInheritedMain,
+    background: theme.colors.backgroundInheritedMain,
+    backgroundHover: theme.colors.backgroundInheritedHover,
+    icon: theme.colors.foregroundInheritedMain,
+  },
+} as const;
+
+const perColorStyle = (color: (typeof labelColors)[number]) => ({
+  "&[data-state=none], &[data-state=hover]": {
+    borderColor: colors[color].border,
+    color: colors[color].icon,
+  },
+  "&[data-state=none]": {
+    background: colors[color].background,
+  },
+  "&[data-state=none]:hover, &[data-state=hover]": {
+    background: colors[color].backgroundHover,
+  },
+  "&[data-state=disabled]": {
+    color: theme.colors.foregroundDisabled,
+  },
+});
+
+const styles = css({
+  display: "flex",
+  width: theme.spacing[11],
+  height: theme.spacing[11],
+  boxSizing: "border-box",
+  alignItems: "center",
+  justifyContent: "center",
+  borderRadius: theme.borderRadius[2],
+  border: "1px solid transparent",
+  variants: {
+    color: {
+      default: perColorStyle("default"),
+      preset: perColorStyle("preset"),
+      local: perColorStyle("local"),
+      remote: perColorStyle("remote"),
+    },
+  },
+  defaultVariants: { color: "default" },
+});
+
+type Props = ComponentProps<"label"> & {
+  css?: CSS;
+  color?: (typeof labelColors)[number];
+  disabled?: boolean;
+  hover?: boolean;
+};
+
+export const NestedIconLabel = forwardRef(
+  (
+    { css, className, color, disabled, hover, ...props }: Props,
+    ref: Ref<HTMLLabelElement>
+  ) => (
+    <label
+      {...props}
+      className={styles({ css, className, color })}
+      data-state={disabled ? "disabled" : hover ? "hover" : "none"}
```

Comment:
state none sounds weird, why not state "default"? figma is using state "default"

Reply:
`none != default` because `&[data-state=default]:hover` would not make sense. For Figma state is either default or hover, there's no "hovered default".

Will try to remove `none` altogether to avoid confusion, but selectors will get more complex probably.